src/test/java/org/apache/commons/collections4/bloomfilter/AbstractCountingBloomFilterTest.java

```diff
@@ -289,6 +296,114 @@ public void testExcludesDuplicates() {
+        bfFrom1.merge(TestingHashers.FROM1);
+        BloomFilter bfFrom11 = new DefaultBloomFilterTest.SparseDefaultBloomFilter(getTestShape());
+        bfFrom11.merge(TestingHashers.FROM11);
+
+        assertEquals(0, bf.getMaxInsert(new IncrementingHasher(0, 1)));
+        assertEquals(0, bf.getMaxInsert(bfFrom0));
+        assertEquals(0, bf.getMaxInsert((BitMapProducer) bfFrom0));
+        assertEquals(0, bf.getMaxInsert((IndexProducer) bfFrom0));
+
+        assertEquals(from1, bf.getMaxInsert(TestingHashers.FROM1));
+        assertEquals(from1, bf.getMaxInsert(bfFrom1));
+        assertEquals(from1, bf.getMaxInsert((BitMapProducer) bfFrom1));
+        assertEquals(from1, bf.getMaxInsert((IndexProducer) bfFrom1));
+
+        assertEquals(from11, bf.getMaxInsert(TestingHashers.FROM11));
+        assertEquals(from11, bf.getMaxInsert(bfFrom11));
+        assertEquals(from11, bf.getMaxInsert((BitMapProducer) bfFrom11));
+        assertEquals(from11, bf.getMaxInsert((IndexProducer) bfFrom11));
+    }
+
+    @Test
+    public void testGetMaxInsert() {
+        CountingBloomFilter bf = createEmptyFilter(getTestShape());
+        verifyMaxInsert(bf, 0, 0);
+        bf.merge(TestingHashers.FROM1);
+        verifyMaxInsert(bf, 1, 0);
+        bf.merge(TestingHashers.FROM1);
+        verifyMaxInsert(bf, 2, 0);
+        bf.merge(TestingHashers.FROM11);
+        verifyMaxInsert(bf, 2, 1);
+        bf.remove(TestingHashers.FROM1);
+        verifyMaxInsert(bf, 1, 1);
+        // verify remove false positive works
+        // Incrementing hasher 5,1 spans the single count cells for both FROM1 and FROM11
+        assertEquals(1, bf.getMaxInsert(new IncrementingHasher(5, 1)));
+        bf.remove(new IncrementingHasher(5, 1));
+        verifyMaxInsert(bf, 0, 0);
+        assertEquals(0, bf.getMaxInsert(new IncrementingHasher(5, 1)));
+    }
+
+    private void assertCell3(CountingBloomFilter bf, int value) {
+        bf.forEachCell((k, v) -> {
+            if (k == 3) {
+                assertEquals(value, v, "Mismatch at position (3) "+k);
+            } else {
+                assertEquals(0, v, "Mismatch at position "+k);
+            }
+            return true;
+        });
+    }
+
+    @Test
+    public void mergeIncrementsAllCellsTest() {
+        CountingBloomFilter f1 = createEmptyFilter(Shape.fromKM(1, 10));
+        CountingBloomFilter f2 = f1.copy();
+        CountingBloomFilter f3 = f1.copy();
+        // index producer produces 3 two times.
+        IndexProducer ip = p -> {
+            p.test(3);
+            p.test(3);
+            return true;
+        };
+        // The merge should increment cell 3 by 1
+        f1.merge(ip);
+        assertCell3(f1, 1);
+
+        // The add should increment cells 3 by 2
+        f2.add(CellProducer.from(ip));
+        assertCell3(f2, 2);
+
+        // This merge will increment by 1 as the round-trip makes the indices unique
```

Comment:
This test is redundant. The round trip does not make the indices unique. So the test is effectively:
```java
f3.merge(ip)
```
This is same as the test using f1.